### ISSUE TITLE
eigenvector centrality - appears to have a bug for directed acyclic networks

### ISSUE THREAD

**Issue by joelmiller** [networkx/networkx#4021]
Please see this:

https://stackoverflow.com/q/62561548/2966723

When I look at `eigenvector_centrality`'s code, it looks like it should break on directed acyclic networks.  In at least this case it gives an answer and I can't understand what that answer means.

**Comment by dschult** [comment]
You are correct that eigenvector_centrality isn't a useful measure for directed acyclic graphs. 
See [this nice description of centrality][1]. The eigenvector centrality should be useless for both the numpy and non-numpy versions of the code. It's better to use a different measure such as Katz or PageRank when the eigenvectors of the largest eigenvalue are not well defined.  

Perhaps the docs should warn people? Suggestions for language?

  [1]: http://www.pitt.edu/~kpele/Materials15/module2.pdf

**Comment by joelmiller** [comment]
I'm really surprised that the algorithm successfully converged on this network.  I don't understand how that happened.  My expectation is that the calculation would give 0 weight to the nodes with no edges in, then 0 weight to those with at most a length 1 edge in, etc.  And then finally it would give 0 weight to everything.  I think the normalization stops that, but I'm surprised it could converge to anything.

I would consider refusing to do this algorithm on a directed network, or else look to see why it managed to converge and test that - I suspect the normalization factor is exploding.
